app/src/main/java/com/antosik/benchproject/app/common/binding/view/ViewBindingAdapters.kt

```diff
@@ -2,8 +2,18 @@ package com.antosik.benchproject.app.common.binding.view
 
 import android.view.View
 import androidx.databinding.BindingAdapter
+import com.antosik.benchproject.app.common.state.UIState
 
 @BindingAdapter("isVisible")
 fun View.setVisibility(flag: Boolean) {
     visibility = if (flag) View.VISIBLE else View.GONE
 }
+@BindingAdapter("showOnLoading")
+fun View.showOnLoading(uiState: UIState) {
+    visibility = if (uiState is UIState.Loading) View.VISIBLE else View.GONE
```

Comment:
Same here, I would rather go with some method on `UIState` instead of if..else in binding adapter. Binding adapters should be as dumb as possible

Reply:
exactly, i would drop this and pass a bool value from function for UIState and of course test the UI state function, and use only Your setVisibility adapter :D

Reply:
I changed it therefore I can use the old adapter `isVisible` :D